src/test_utils/public/enzyme_helpers.tsx

```diff
@@ -128,3 +128,15 @@ export function renderWithIntl<T>(
 }
 
 export const nextTick = () => new Promise(res => process.nextTick(res));
+
+export function shallowWithI18nProvider<T>(child: ReactElement<T>) {
```

Comment:
Did you try using this? https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#unit-tests

If so and it didn't work, lets talk about it. I want to make sure our reasoning is solid before adding new enzyme helpers.

Reply:
It seems that the doc you're talking about is about using `shallowWithIntl()` or `mountWithIntl()` when `injectI18n` HOC is used.

We cannot use them in this PR, because I removed them. And it is recommended to do so in [the i18n doc](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md). 

As far as I know, we need `withI18nProvider` functions to avoid the error message.

Reply:
And as you can see [here in the commit](https://github.com/elastic/kibana/pull/44043/commits/46cba3484aee17b77a0b4708b24c05d5d64d1ee0), `WrapperComponent`s are gone.

Reply:
I think we should update https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/GUIDELINE.md#unit-tests so it doesn't use `props.intl` in the example.  I think @sainthkh's approach may be the right one, as long as there is a `FormattedMessage` somewhere in the line, then it'll need the provider wrapper.